tests/snappi_tests/utils/metrics.py

```diff
@@ -0,0 +1,116 @@
+"""
+This file defines the classes receiving metrics from snappi tests and processing them.
+"""
+import logging
+import json
+import datetime
+import time
+
+from copy import deepcopy
+from pprint import pprint
+from typing import Dict, Final, List, Union
+
+# Only certain labels are allowed
+METRIC_LABEL_TEST_TESTBED: Final[str] = "test.testbed"
+METRIC_LABEL_TEST_BUILD: Final[str] = "test.os.version"
+METRIC_LABEL_TEST_CASE: Final[str] = "test.testcase"
+METRIC_LABEL_TEST_FILE: Final[str] = "test.test_file"
+METRIC_LABEL_TEST_JOBID: Final[str] = "test.job_id"
+METRIC_LABEL_DEVICE_ID: Final[str] = "device.id"
+METRIC_LABEL_DEVICE_PORT_ID: Final[str] = "device.port.id"
+METRIC_LABEL_DEVICE_PSU_ID: Final[str] = "device.psu.id"
+METRIC_LABEL_DEVICE_QUEUE_ID: Final[str] = "device.queue.id"
+METRIC_LABEL_DEVICE_SENSOR_ID: Final[str] = "device.sensor.id"
+
+
+class PeriodicMetricsReporter:
+    def __init__(self, resource_labels: Dict[str, str]):
+        # Will be replaced with a real initializer such as OpenTelemetry
+        self.resource_labels = deepcopy(resource_labels)
+        self.metrics = []
+
+    def stash_record(self, new_metric: 'Metric', labels: Dict[str, str], value: Union[int, str, float]):
```

Comment:
do we allow value to be a str?

Reply:
Yes, if we want to track things like sensor warning status, power LED status.

Reply:
emmmm, how would you like to pass them or store them? iirc, the data point in OTEL only support numbers.